common/src/main/java/com/wynntils/services/itemfilter/ItemFilterService.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import com.wynntils.core.components.Service;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+public class ItemFilterService extends Service {
+    private final Map<String, Function<String, ? extends ItemFilter>> filterSuppliers = new HashMap<>();
```

Comment:
Also, the prepare() method seem to be an artifact of this design choice, rather than something inherent in the problem domain.

So if you move the validation and parsing of the string into a ItemFilterFactory, I think you can get a simpler design. You can have something along these lines:
```
interface ItemFilterFactory {
  ItemFilter create(String inputString);

  String getKeyword();

  String getI18nKey();
}

interface ItemFilter {
   boolean matches(WynnItem wynnItem);
}
```

and you can basically turn the current level and profession item filters into item filter factories, and just have a trivial nested class that implements their respective ItemFilter implementations.

And then you just do something like:
```
       registerFilterFactory(new  LevelItemFilterFactory());
       registerFilterFactory(new  ProfessionIItemFilterFactory());
```
in the Service.